components/menu/Menu.js

```diff
@@ -17,162 +17,191 @@ const POSITION = {
 };
 
 class Menu extends PureComponent {
-  state = {
-    active: this.props.active,
-  };
+  state = {};
 
   componentDidMount() {
-    this.positionTimeoutHandle = setTimeout(() => {
-      const { width, height } = this.menuNode.getBoundingClientRect();
-      const position = this.props.position === POSITION.AUTO ? this.calculatePosition() : this.props.position;
-      this.setState({ position, width, height });
-    });
+    const { width, height } = this.menuNode.getBoundingClientRect();
+    this.setState({ width, height });
+
+    const { position } = this.props;
+    if (position === POSITION.AUTO) {
+      this.setState({ position: this.calculatePosition });
```

Comment:
Q: Is this correct? Now you're assigning a function to `position`, right? While you want to have the calculated number? So I'd say this needs to be 
```js
this.setState({ position: this.calculatePosition() });
```

Reply:
You actually forgot to change it here?